src/dataaccess/McodeV09SmartOnFhirDataSource.js

```diff
@@ -108,7 +142,29 @@ class McodeV09SmartOnFhirDataSource extends IDataSource {
                     queries.push(result);
                 }
 
-                return Promise.all(queries);
+                const referenceQueries = [];
+                let nonEmptyResources = [];
+                return Promise.all(queries)
+                    .then((resources) => {
+                        // After the first pass of queries are resolved, fetch any additional resources that were
+                        // referenced that have not yet been fetched.
+
+                        // Filter out any searchset bundles that don't have any entries
+                        nonEmptyResources = nonEmptyResources.concat(resources.filter(res => res.entry));
+                        this.referencedEntries.forEach((key) => {
+                            const referenceArray = key.split('/');
+                            const resourceType = referenceArray[0];
+                            const resourceId = referenceArray[1];
```

Comment:
I don't think references are guaranteed to always be of the format `<resource type/<id>`. For example, they could be full URLs, like `http://<fhir server>/<resource type>/<id>`. In this case, doing a split on `'/'` would not work they way it's intended to attempt to parse out the resource type and the id from a reference.

Not sure of the best way to approach being smarter about this. We could use a regex, or always just look at the things prior to and after the last slash? Not sure.